test/suite/hook_tiered.py

```diff
@@ -0,0 +1,145 @@
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+# [TEST_TAGS]
+# ignored_file
+# [END_TAGS]
+
+# hook_tiered.py
+#
+# Substitute tiered tables for regular (row-store) tables in Python tests.
+#
+# These hooks can be used to run the existing cursor tests on tiered tables.
+# They identify tests that create row-store tables and create tiered tables
+# instead.  The hook takes an optional argument to specify how many tiers
+# to create. The default is 2.
+#
+# To run with 3 tiers per table:
+#     ../test/suite/run.py --hooks tiered=3 cursor
+#
+# The hooks work with may other tests in the python suite but also encounter
+# a variety of failures that I haven't tried to sort out.
+from __future__ import print_function
+
+import os, sys, wthooks
+import unittest
+from wttest import WiredTigerTestCase
+
+# Print to /dev/tty for debugging, since anything extraneous to stdout/stderr will
+# cause a test error.
+def tty(s):
+    WiredTigerTestCase.tty(s)
+
+# These are the hook functions that are run when particular APIs are called.
+
+# Called to replace Session.create
+def session_create_replace(arg, orig_session_create, session_self, uri, config):
+    if config == None:
+        base_config = ""
+    else:
+        base_config = config
+
+    # If the test is creating a table (not colstore or lsm), create a tiered table instead,
+    # using arg to determine number of tiers. Otherwise just do the create as normal.
+    #
+    # NOTE: the API for creating a tiered table will change in WT-7173 and this will need to
+    # be updated.
+    if uri[:6] == "table:" and "key_format=r" not in base_config and "type=lsm" not in base_config:
+        tier_string = ""
+        for i in range(arg):
+            new_uri = uri.replace('table:', 'file:tier' + str(i) + '_')
+            ret = orig_session_create(session_self, new_uri, config)
+            if ret != 0:
+                return ret
+            tier_string = tier_string + '"' + new_uri + '", '
+        tier_config = 'type=tiered,tiered=(tiers=(' + tier_string[0:-2] + ')),' + base_config
+        WiredTigerTestCase.verbose(None, 3,
+            'Creating tiered table {} with config = \'{}\''.format(uri, tier_config))
+        ret = orig_session_create(session_self, uri, tier_config)
+    else:
+        ret = orig_session_create(session_self, uri, config)
+    return ret
+
+# Called to replace Session.drop
+def session_drop_replace(arg, orig_session_drop, session_self, uri, config):
+    # Drop isn't implemented for tiered tables.  Only do the delete if this could be a
+    # uri we created a tiered table for.  Note this isn't a precise when we did/didn't
+    # create a tiered table, but we don't have the create config around to check.
```

Comment:
If we cared, we could probably check for the existence of the tiered files to know.  But if things are working well enough, there's no need to go to the trouble.